addons/flashdrive/functions/fnc_initInterface.sqf

```diff
@@ -4,79 +4,84 @@
+params ["_device", "_config", "_parentActionName"];
 
-private _flash_drives = {
-	params ["_target", "_player", "_params"];
-	_params params ["_config"];
-
-	private _actions = [];
-	private _class = configFile >> "CfgWeapons" >> "Item_FlashDisk_AE3";
-
-	private _add = {
+if(!isDedicated) then
+{
+	private _parentActionPath = [_device, _parentActionName] call AE3_main_fnc_interaction_addParentAction;
+	
+	private _flash_drives =
+	{
 		params ["_target", "_player", "_params"];
-		([_target, _player] + _params) remoteExec ["AE3_flashdrive_fnc_connectFlashDrive", 2];
-	};
+		_params params ["_config"];
 
-	_config params ['_index', '_name', '_rel_pos', '_rot_yaw', '_rot_pitch', '_rot_roll'];
-	{
-		private _action = [_name, (getText (configFile >> "CfgWeapons" >> _x >> "displayName")), "", _add, {true}, {}, [_x, _config]] call ace_interact_menu_fnc_createAction; 
-		_actions pushBack [_action, [], _target];
-	} forEach (items _player select {inheritsFrom (configFile >> "CfgWeapons" >> _x) == _class});
+		private _actions = [];
+		private _class = configFile >> "CfgWeapons" >> "Item_FlashDisk_AE3";
 
-	_actions;
-};
+		private _add = {
+			params ["_target", "_player", "_params"];
+			([_target, _player] + _params) remoteExec ["AE3_flashdrive_fnc_connectFlashDrive", 2];
+		};
 
-private _children = {
-	params ["_target", "_player", "_params"];
-	_params params ['_flash_drives'];
+		_config params ["_index", "_name", "_rel_pos", "_rot_yaw", "_rot_pitch", "_rot_roll"];
+		{
+			private _action = [_name, (getText (configFile >> "CfgWeapons" >> _x >> "displayName")), "", _add, {true}, {}, [_x, _config]] call ace_interact_menu_fnc_createAction; 
+			_actions pushBack [_action, [], _target];
+		} forEach (items _player select {inheritsFrom (configFile >> "CfgWeapons" >> _x) == _class});
 
-	private _interfaces = _target getVariable "AE3_USB_Interfaces";
-	private _occupiedList = _target getVariable "AE3_USB_Interfaces_occupied";
+		_actions;
+	};
 
-	private _actions = [];
-	
-	private _condition = 
+	private _children =
 	{
 		params ["_target", "_player", "_params"];
-		(_params select 0) params ['_index', '_name', '_rel_pos', '_rot_yaw', '_rot_pitch', '_rot_roll'];
+		_params params ["_flash_drives"];
+
+		private _interfaces = _target getVariable "AE3_USB_Interfaces";
 		private _occupiedList = _target getVariable "AE3_USB_Interfaces_occupied";
 
-		isNull (_occupiedList select _index);
-	};
-	
-	{
-		_y params ['_index', '_name', '_rel_pos', '_rot_yaw', '_rot_pitch', '_rot_roll'];
+		private _actions = [];
 		
-		if (isNull (_occupiedList select _index)) then 
+		private _condition = 
 		{
-			private _action = [_name, _name, "", {}, _condition, _flash_drives, [_y]] call ace_interact_menu_fnc_createAction;  
-			_actions pushBack [_action, [], _target];
-		};
-	} forEach _interfaces;
-
-	_actions;
-};
+			params ["_target", "_player", "_params"];
+			(_params select 0) params ["_index", "_name", "_rel_pos", "_rot_yaw", "_rot_pitch", "_rot_roll"];
```

Comment:
```suggestion
			private _driveInfo = _params select 0;
			if !(_driveInfo isEqualType []) then {
				_driveInfo = _params select 1;
			};
			_driveInfo params ["_index", "_name", "_rel_pos", "_rot_yaw", "_rot_pitch", "_rot_roll"];
```
This change fixes #414, details [here](https://github.com/y0014984/Advanced-Equipment/issues/414#issuecomment-2052427187).

Reply:
Thanks to Dystopian's https://github.com/acemod/ACE3/pull/9946, this rough fix won't be necessary, the next ACE3 Patch version will resolve the issue.

Reply:
This has now been fixed publicly by [ACE v3.17.1](https://github.com/acemod/ACE3/releases/tag/v3.17.1).